example_problems/nonrelativistic_boltzmann/linear_modes_test/perturbed_temperature_tests/1D1V/collisional_damping/initialize.py

```diff
@@ -0,0 +1,79 @@
+"""
+Functions which are used in assigning the I.C's to
+the system.
+"""
+
+import arrayfire as af
+import numpy as np
+
+def initialize_f(q1, q2, p1, p2, p3, params):
+
+    m = params.mass_particle
+    k = params.boltzmann_constant
+
+    rho_b = params.rho_background
+    T_b = params.temperature_background
+
+    p1_bulk = params.p1_bulk_background
+    p2_bulk = params.p2_bulk_background
+    p3_bulk = params.p3_bulk_background
+
+    pert_real = params.pert_real
+    pert_imag = params.pert_imag
+
+    k_q1 = params.k_q1
+    k_q2 = params.k_q2
+
+    # Calculating the perturbed density:
+    T = T_b + (pert_real * af.cos(k_q1 * q1 + k_q2 * q2) -
+               pert_imag * af.sin(k_q1 * q1 + k_q2 * q2)
+               )
+
+    # Depending on the dimensionality in velocity space, the
+    # distribution function is assigned accordingly:
+    if (params.p_dim == 3):
+
+        f = rho_b * (m / (2 * np.pi * k * T))**(3 / 2) * \
+            af.exp(-m * (p1 - p1_bulk)**2 / (2 * k * T)) * \
+            af.exp(-m * (p2 - p2_bulk)**2 / (2 * k * T)) * \
+            af.exp(-m * (p3 - p3_bulk)**2 / (2 * k * T))
+
+    elif (params.p_dim == 2):
+
+        f = rho_b * (m / (2 * np.pi * k * T)) * \
+            af.exp(-m * (p1 - p1_bulk)**2 / (2 * k * T)) * \
+            af.exp(-m * (p2 - p2_bulk)**2 / (2 * k * T))
+
+    else:
+
+        f = rho_b *\
+            af.sqrt(m / (2 * np.pi * k * T)) * \
+            af.exp(-m * (p1 - p1_bulk)**2 / (2 * k * T))
+
+    af.eval(f)
+    return (f)
+
+
+# In case of user-defined fields the initial-conditions are provided
+# using the functions below. It should be ensured that these satisfy
+# maxwell's constraint equations.(add check - TODO)
+
+
+def initialize_E(q1, q2, p1, p2, p3, params):
+    # Declare initial conditions for E here
+    E1 = 0 * q1[:, :, 0]
+    E2 = 0 * q1[:, :, 0]
+    E3 = 0 * q1[:, :, 0]
+
```

Comment:
What does `q1[:, :, 0]` indicate? `Nq2 x Nq1` ?

Reply:
Oh. That needs to be changed to:
```python
E1 = 0 * q1
E2 = 0 * q1
E3 = 0 * q1
```
That was with the older convention, where `q1` was of shape `(Nq1, Nq2, Np1*Np2*Np3)`. Now since all batched operations are taken care of using `af.broadcast`, `q1` and `q2` are now of shape `(N_q1, N_q2)`